# 04 — Model Evaluation
Week 5 baseline results

In [1]:
import sys
sys.path.insert(0, '..')

from src.models.train import train_baseline_model
from src.models.evaluate import evaluate_model

pipeline, metadata, (X_test, y_test) = train_baseline_model()
result = evaluate_model(pipeline, X_test, y_test)
print(result.summary())

2026-09-08 07:27:56,539 [INFO] Loaded 5000 rows, 18 columns from /home/claude/healthconnect-ml/data/raw/HealthConnect_Appointment_Data.csv


2026-09-08 07:27:56,565 [INFO] Dropped 263 rows with outcome 'Cancelled' (5000 -> 4737 rows)


2026-09-08 07:27:56,579 [INFO] Training Logistic Regression baseline on 3789 rows


HealthConnect Baseline Model Evaluation
Recall (No-Show)   [PRIMARY]: 0.676
Precision (No-Show)          : 0.612
F1-score                     : 0.643
ROC-AUC                      : 0.667
Accuracy                     : 0.615

Confusion Matrix (rows=actual, cols=predicted; order [Attended, No-Show]):
  TN=255  FP=208
  FN=157  TP=328

False negatives (missed no-shows): 157 — this is the primary error to minimize.


## Interpretation
Recall (No-Show) is the primary metric: 157 false negatives on the held-out test set means 157 appointments that actually no-showed were predicted as Attended — the exact failure this system exists to reduce. This baseline catches roughly two-thirds of actual no-shows. Week 6 will compare Random Forest and Gradient Boosting against this baseline.